File: 8kyu/(8 kyu) Count of positives - sum of negatives/(8 kyu) Count of positives - sum of negatives.java
// 1 Plain Solution
public class Kata {
  public static int[] countPositivesSumNegatives(int[] input) {
    if (input == null || input.length == 0) {
      return new int[] {}
    };

    int count = 0, sum = 0;

    for (int i : input) {
      if (i > 0) {
        count ++;
      }
      if (i < 0) {
        sum += i;
      }
    }

    return new int[] { count, sum };
  }
}

// 2 Optimized solution
public class Kata {
  public static int[] countPositivesSumNegatives(int[] input) {    
    if (input == null || input.length == 0) {
      return new int[] {};
    }
    
    int[] res = { 0, 0 };

    for (int x: input) {
      if (x > 0) { res[0]++; } else { res[1] += x; }
    }
    
    return res;
  }
}

// 3 Clever solution
import java.util.stream.*;

public class Kata {
  public static int[] countPositivesSumNegatives(int[] input) {
    return input == null || input.length == 0 ? 
      new int[0] : 
      new int[] {
        (int)IntStream.of(input).filter(i -> i > 0).count(),
        IntStream.of(input).filter(i -> i < 0).sum()
      };
  }
}

File: 8kyu/(8 kyu) Count of positives - sum of negatives/(8 kyu) Count of positives - sum of negatives.py
# #1
# def count_positives_sum_negatives(arr):
#     if len(arr) == 0:
#         return []
#     res = [0, 0]
#     for x in arr:
#         if x > 0:
#             res[0] += 1
#         else:
#             res[1] += x
#     return res
# #2
def count_positives_sum_negatives(arr):
    pos = sum(1 for x in arr if x > 0)
    neg = sum(x for x in arr if x < 0)
    return [pos, neg] if len(arr) else []
